superset/sql_lab.py

```diff
@@ -490,9 +490,9 @@ def execute_sql_statements(  # pylint: disable=too-many-arguments, too-many-loca
                     ex, query, session, payload, prefix_message
                 )
                 return payload
-
         # Commit the connection so CTA queries will create the table.
-        conn.commit()
+        if apply_ctas:
+            conn.commit()
```

Comment:
I think this is fine, and the `commit` might not even be needed at all (but does no harm) since we close the connection after (exiting the context manager) and open a new one when running `select_star`.

(I wonder if the `commit` was needed before we added the `closing` context manager...)